Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,655 @@
+                                                        testStepLog.ActualResult = string.Empty;
+                                                        SetTestStepLogStatus(matchingActivity, ref testStepLog, statuses);
+                                                        stepsStatuses.Add(testStepLog.Status);
+                                                        testStepLogs.Add(testStepLog);
+                                                        testStepsCount++;
+                                                    }
+                                                }
+                                            }
+                                            else
+                                            {
+                                                Activity matchingActivity = activities.Where(x => x.ExternalID == step.Id.ToString()).FirstOrDefault();
+                                                if (matchingActivity != null)
+                                                {
+                                                    QTestApiModel.TestStepLogResource testStepLog = new QTestApiModel.TestStepLogResource(null, step.Id);
+                                                    testStepLog.ActualResult = string.Empty;
+                                                    SetTestStepLogStatus(matchingActivity, ref testStepLog, statuses);
+                                                    stepsStatuses.Add(testStepLog.Status);
+                                                    testStepLogs.Add(testStepLog);
+                                                    testStepsCount++;
+                                                }
+                                            };
+                                        }                                        
+
+                                        //update the TC general status based on the activities status collection.                                
+                                        if (stepsStatuses.Where(x => x.Name == "Failed").Count() > 0)
+                                            testCaseStatus = statuses.Where(z => z.Name == "Failed").FirstOrDefault();
+                                        else if (stepsStatuses.Where(x => x.Name == "No Run").Count() == testStepsCount || stepsStatuses.Where(x => x.Name == "Not Applicable").Count() == testStepsCount)
+                                            testCaseStatus = statuses.Where(z => z.Name == "Unexecuted").FirstOrDefault();
+                                        else if (stepsStatuses.Where(x => x.Name == "Passed").Count() == testStepsCount || (stepsStatuses.Where(x => x.Name == "Passed").Count() + stepsStatuses.Where(x => x.Name == "Not Applicable").Count()) == testStepsCount)
+                                            testCaseStatus = statuses.Where(z => z.Name == "Passed").FirstOrDefault();
+                                        else
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)